config/config.go

```diff
@@ -172,6 +170,23 @@ func enrichFromEnvvars(config *HorizonConfig) error {
 		config.Edge.ExchangeMessageDynamicPoll = false
 	}
 
+	if apiPort := os.Getenv(AnaxAPIPort); apiPort != "" {
+		if config.Edge.APIListen != "" {
+			listen := strings.Split(config.Edge.APIListen, ":")
+			if len(listen) == 2 {
+				config.Edge.APIListen = fmt.Sprintf("%v:%v", listen[0], apiPort)
+			} else {
+				config.Edge.APIListen = fmt.Sprintf("127.0.0.1:%v", apiPort)
+			}
+		} else {
+			config.Edge.APIListen = fmt.Sprintf("127.0.0.1:%v", apiPort)
+		}
+	} else {
+		if config.Edge.APIListen == "" {
+			config.Edge.APIListen = fmt.Sprintf("127.0.0.1:%v", AnaxAPIPortDefault)
+		}
```

Comment:
how about the case when config.Edge.APIListen != "" but len(listen) ==1?  It should be `fmt.Sprintf("%v:%v", listen[0], AnaxAPIPortDefault)`

Reply:
Please also change  the 2 lines here https://github.com/open-horizon/anax/blob/master/anax-in-container/horizon-container#L185-L188. Change 80 to 8510

Reply:
Will update horizon-container.
For the Edge.APIListen change, see above, not sure how it should work because we've never validated that field before, it's hard for the user to change and we set it to a default value.